Q14/Codeville.rb

```diff
@@ -0,0 +1,18 @@
+def solution(weights, limit)
+    # p weights.sort!
+    boats = 0
+    check = 0
+    index = 0
+  
+    weights.each do |x|
+      check += weights[index] 
+      if check + weights[index] >= 200
+        boats += 1
+        check = 0
+      end
+      index += 1
+    end
+    p boats
+  end
+  
```

Comment:
Hey does this algorithm give correct results for test cases? Here are a few
`class MinBoatTest < Minitest::Test
  def test_one_boat_needed
    assert_equal 1, min_boats([70, 70], 200)
  end

  def test_over_capacity_under_weight
    assert_equal 2, min_boats([70, 50, 70], 200)
  end

  def test_over_weight_under_capacity
    assert_equal 2, min_boats([200, 200], 200)
  end

  def test_even_number_of_people
    assert_equal 3, min_boats([100, 200, 150, 80], 200)
  end

  def test_odd_number_of_people
    assert_equal 3, min_boats([100, 80, 30, 150, 80], 200)
  end
end`

Reply:
All of them pass barring the last one, I am having trouble with getting the last one to work

Reply:
The test all pass now